third_party/gems/rules.bzl

```diff
@@ -0,0 +1,249 @@
+load("//third_party/gems:gemfile.bzl", _parse_gemfile_lock = "parse_gemfile_lock")
+
+def _label_to_path(label):
+    """
+    Given a Label, turn it into a path by keeping the directory part of the
+    name, and attaching that to the package path.
+    """
+
+    base = label.package
+
+    name = label.name
+
+    last_slash = name.rfind("/")
+    if last_slash >= 0:
+        base = "{}/{}".format(base, name[0:last_slash])
+
+    return base
+
+def _read_file(repo_ctx, label):
+    """
+    Bazel-0.21.0 lacks `repo_ctx.read`, so this simulates it by briefly
+    symlinking the file into the current repo, cat-ing it, and then removing
+    the symlink.
+    """
+
+    if repo_ctx.attr.debug:
+        print("Reading {}".format(label))
+
+    repo_ctx.symlink(label, "tmp")
```

Comment:
what are isolation guarantees here. Can this race with other version of this command?

Reply:
I think that there's no race here, as it's only used to sequentially read the `Gemfile.lock` files passed in via `gemfile_locks`.

Reply:
👍